Tests/Client/Unit Tests/WebSocket/WebSocketTests.swift

```diff
@@ -118,6 +118,29 @@ final class WebSocketTests: XCTestCase {
         time.run(numberOfSeconds: 3 * pingInterval)
         XCTAssertEqual(socketProvider.sendPingCalledCounter, 1 + 3)
     }
+    
+    func test_typingEvent_sendStopTypingonTimeout() {
+        let expectation = self.expectation(description: "A stop typing event on timeout")
+        
+        _ = webSocket.subscribe { event in
+            if case .connectionChanged(let state) = event, state.isConnected {
+                self.socketProvider.simulateMessageReceived(["type": "typing.start",
+                                                           "user": ["id": "u1",
+                                                                    "role": "user",
+                                                                    "created_at": "2020-05-12T12:42:56.450979Z",
+                                                                    "updated_at": "2020-05-12T12:42:56.450979Z",
+                                                                    "online": true]])
+            }
+            
+            if case .typingStop(let user, _, _) = event, user.id == "u1" {
+                expectation.fulfill()
+            }
+        }
+        
+        test_connectionFlow()
+        time.run(numberOfSeconds: 31) // 30 sec for timeout.
+        wait(for: [expectation], timeout: 5)
```

Comment:
Why do you wait 5 seconds here?

Reply:
It won't wait 5 sec if the expectation will be fulfilled. It's actually the default time.